pwp/commands/install.py

```diff
@@ -0,0 +1,60 @@
+import os
+import sys
+from pip._internal.cli.main import main as pip_main
+import importlib.metadata
+
+
+def get_installed_package_version(package_name):
+    """Get the installed version of a package."""
+    try:
+        return importlib.metadata.version(package_name)
+    except importlib.metadata.PackageNotFoundError:
+        return None
+
+
+def create_or_update_requirements(packages: dict[str, str | None]):
```

Comment:
`create_or_update_requirements` now takes packages as a `dict<name, version>` instead of a `list`, where `version` can be null.